mempool/src/shared_mempool/peer_manager.rs

```diff
@@ -127,34 +127,35 @@ impl PeerManager {
         let is_new_peer = !peer_states.contains_key(&peer);
         if self.is_upstream_peer(&peer, Some(metadata.origin)) {
             counters::active_upstream_peers(&peer.raw_network_id()).inc();
-            if peer.raw_network_id() == NetworkId::Validator {
-                // For a validator network, resume broadcasting from previous state.
-                // We can afford to not re-broadcast here since the transaction is already in a validator.
-                if is_new_peer {
-                    peer_states.insert(peer, PeerSyncState::new(metadata));
-                } else if let Some(peer_state) = peer_states.get_mut(&peer) {
-                    peer_state.is_alive = true;
-                    peer_state.metadata = metadata;
-                }
-            } else {
-                // For a non-validator network, potentially re-broadcast any transactions that have not been
-                // committed yet.
-                // This is to ensure better reliability of transactions reaching the validator network.
+            // If we have a new peer, let's insert new data, otherwise, let's just update the current state
+            if is_new_peer {
                 peer_states.insert(peer, PeerSyncState::new(metadata));
+            } else if let Some(peer_state) = peer_states.get_mut(&peer) {
+                peer_state.is_alive = true;
+                peer_state.metadata = metadata;
             }
         }
         drop(peer_states);
         self.update_failover();
         is_new_peer
     }
 
+    /// Disables a peer if it can be restarted, otherwise removes it
     pub fn disable_peer(&self, peer: PeerNetworkId) {
-        if let Some(state) = self.peer_states.lock().get_mut(&peer) {
-            counters::active_upstream_peers(&peer.raw_network_id()).dec();
-            // TODO: What about garbage collection?
-            state.is_alive = false;
+        // Validators can be restarted ata  later time
+        // TODO: Determine why there's this optimization
```

Comment:
I would just throw this against a cluster test that supports up/down

also now that we're exploring this space, it might be worthwhile to keep track of who sent you what. I don't think we should be forwarding VFN failover to other VFNs -- it should only go to the validator...

Reply:
One problem at a time :)

Turns out this change causes the legacy failover test to break, so I'm trying to workout that one first that's why it's a `draft`